google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala

```diff
@@ -66,7 +67,7 @@ private final class GooglePubSubSource(projectId: String,
               }
             case Nil =>
               state = Fetching
-              fetch(mat)
+              mat.scheduleOnce(1.second, () => fetch(mat))
```

Comment:
This will schedule `fetch` to be ran on the execution context that is different from the graph stage context. Instead it should use `scheduleOnce` from the `TimerGraphStageLogic` ([more info in the docs here](https://doc.akka.io/docs/akka/current/stream/stream-customize.html#using-timers)).

It is important because inside of `fetch` it updates a mutable `state` variable. And we need to make sure that all accesses to that variable are done from the graph stage context.

Reply:
Thanks for taking the time to point me in the right direction -- greatly appreciated!

I've updated the PR to make use of TimerGraphStageLogic as suggested.